osi_groundtruthinitconfiguration.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto2";
+
+option optimize_for = SPEED;
+
+import "osi_common.proto";
+import "osi_version.proto";
+
+package osi3;
+
+//
+// \brief The configuration settings for splitting static and dynamic 
+// content for performance reasons. Static content of \c GroundTruth 
+// can be sent at initialization, dynamic content during simulation 
+// as \c SensorView. What is regarded as static content is described here. 
+//
+message GroundTruthInitConfiguration
+{
+    // The interface version used by the sender (simulation environment).
+    //
+    optional InterfaceVersion version = 1;
+
+    // Omit static street layer
+    //
+    // The street layer, i.e. \c Lane, \c LaneBoundary and \c RoadMarking 
+    // is regarded static and should be sent only at initialization.
+    //
+    optional bool omit_static_street_layer = 2;
+
+    // Omit static objects
+    //
+    // Objects of type \c MovingObject and \c StationaryObject which do not 
+    // have changing attributes (e.g. changing position) during a simulation 
+    // are regarded static and should be sent only at initialization. 
+    //
+    // \note: While sending certain object at initialization and others 
+    // at runtime, it must be ensured that each object still has a unique id.
+    //
+    optional bool omit_static_objects = 3;    
```

Comment:
I think this is a sensible definition, but will need matching changes to OSMP. Currently https://github.com/OpenSimulationInterface/osi-sensor-model-packaging/blob/master/doc/specification.rst says:

> `OSMPGroundTruthInit` MUST contain all static data (e.g. roads) encountered by the model during a simulation run. Any dynamic data (e.g. MovingObjects) it contains MUST NOT be used and has no specified semantics.

If I remember correctly, the second sentence was added on request of dSpace, who said they could not guarantee that only static objects would be present in the GroundTruthInit message. (The change was https://github.com/OpenSimulationInterface/osi-sensor-model-packaging/pull/56/commits/cf890b4d47d1156e3a25e36d87c19ded758ea64c, in case you want to look up the original discussion in SETLevel4to5). I understand where this argument is coming from (probably most people just use an already existing function to generate the GroundTruth, add an object to the world, set the radius to maximum, and generate the GroundTruth from there), but I do not find it particularly compelling: it's quite easy to remove the dynamic objects before passing them to an FMU as part of `OSMPGroundTruthInit`.

If we allow to `omit static objects` (which I find important), then OSMP has to be changed to say that the `OSMPGroundTruthInit` message may only contain completely static objects, which will not appear in later messages. If any dynamic object (which can change later or disappear) is present in the initial ground truth, then it becomes impossible to later determine whether the object is static (and thus not present in a later sensor view), or whether it moved out of sight.

Note: I do find `omit static objects` to be important: our maps typically have 1000s of static objects (trees, poles, patches on the roads, buildings, ...). Example: the map I demoed on the "SETLevel4to5 Halbzeitevent" contains 3574 static objects on the 1x1km map. I had to disable transmitting any static objects (even just within the 200m radius of the pedestrian) to get acceptable performance. So this flag is quite crucial to get realtime performance.

Reply:
Ah, I see now that this is addressed in https://github.com/OpenSimulationInterface/osi-sensor-model-packaging/pull/69 .